internal/cli/cmd/dac/build/build.go

```diff
@@ -0,0 +1,206 @@
+			return outputErr
+		}
+	} else {
+		// Put explicitely a value when not provided, as we use it for file generation in Execute()
+		o.Output = output.YAMLOutput
+	}
+
+	if o.Mode != modeFile && o.Mode != modeStdout {
+		return fmt.Errorf("invalid mode provided: must be either `file` or `stdout`")
+	}
+
+	return nil
+}
+
+func (o *option) Validate() error {
+	if o.File != "" {
+		return o.FileOption.Validate()
+	}
+	return o.DirectoryOption.Validate()
+}
+
+func (o *option) Execute() error {
+	if o.File != "" {
+		return o.processFile(o.File)
+	}
+	// Else it's a directory, thus walk it and process each file
+	err := filepath.Walk(o.Directory, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// Process regular files only
+		if !info.IsDir() {
+			err = o.processFile(path)
+			if err != nil {
+				// Tell the user about the error but don't stop the processing
+				fmt.Printf("error processing file %s: %v\n", path, err)
+			}
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return fmt.Errorf("error processing directory %s: %v", o.Directory, err)
+	}
+
+	return nil
+}
+
+func (o *option) processFile(file string) error {
+	// NB: this command is mostly based on the `eval` command of the cue CLI.
+	// NB2: Since cue is written in Go, we could consider relying on its code instead of going the exec way.
+	//      However the cue code is (for now at least) not well packaged for such external reuse.
+	//      See https://github.com/cue-lang/cue/blob/master/cmd/cue/cmd/eval.go#L87
+	cmd := exec.Command("cue", "eval", file, "--out", o.Output, "--concrete") // #nosec
+
+	// Capture the output of the command
+	cmdOutput, err := cmd.Output()
+	if err != nil {
+		if exitErr, ok := err.(*exec.ExitError); ok {
+			return fmt.Errorf("failed to build %s: %s", file, string(exitErr.Stderr))
+		}
+		return err
+	}
+
+	// If mode = stdout, print the command result on the standard output & don't go further
+	if o.Mode == modeStdout {
+		return output.HandleString(o.writer, string(cmdOutput))
+	}
+	// Otherwise, create an output file under the "built" directory:
+
+	// Create the folder (+ any parent folder if applicable) where to store the output
+	err = os.MkdirAll(filepath.Join(outputFolderName, filepath.Dir(file)), os.ModePerm)
+	if err != nil {
+		return fmt.Errorf("error creating the output folder: %v", err)
+	}
+
+	// Build the path of the file where to store the command output
+	outputFilePath := buildOutputFilePath(file, o.OutputOption.Output)
+
+	// Write the output to the file
+	err = writeToFile(outputFilePath, cmdOutput)
+	if err != nil {
+		return fmt.Errorf("error writing to %s: %v", outputFilePath, err)
+	}
+
+	return output.HandleString(o.writer, fmt.Sprintf("Succesfully built %s at %s", file, outputFilePath))
+}
+
+// buildOutputFilePath generates the output file path based on the input file path
+func buildOutputFilePath(inputFilePath string, ext string) string {
```

Comment:
```suggestion
func (o *option) buildOutputFilePath(inputFilePath string) string {
}
```

I believe this function should belong to the `option` struct since it's using the attribute `OutputOption`